cmd/zfs/zfs_main.c

```diff
@@ -8729,6 +8740,55 @@ main(int argc, char **argv)
 	return (ret);
 }
 
+/*
+ * zfs userns add|del nsnum filesystem
+ *
+ * Add or delete the given dataset to/from the namespace.
+ */
+#ifdef __linux__
+static int
+zfs_do_userns(int argc, char **argv)
+{
+	zfs_handle_t *zhp;
+	unsigned long nsnum;
+	int ret;
+	int attach;
+
+	if (argc < 4) {
+		(void) fprintf(stderr, gettext("missing argument(s)\n"));
+		usage(B_FALSE);
+	}
+	if (argc > 4) {
+		(void) fprintf(stderr, gettext("too many arguments\n"));
+		usage(B_FALSE);
+	}
+
+	if (strcmp(argv[1], "add") == 0) {
+		attach = 1;
+	} else if (strcmp(argv[1], "del") == 0) {
```

Comment:
So the analog on FreeBSD is 'jail' and 'unjail'. Maybe attach and detach? The difference is that it can technically be delegated to more than one namespace.

Reply:
Attach/detach seems sensible to me.